### ISSUE TITLE
Question about sigmaLoss with DSNeRF paper.

### ISSUE THREAD

**Issue by jhq1234** [dunbar12138/DSNeRF#74]
Hi,
Thank you for sharing your great code.
I'm glad to learn about DSNeRF with your kind and detailed explanation

I have a question about sigmaLoss with paper.

I guess sigma loss is the ray distribution loss in the paper notation by h(t).
From the paper, h(t) = T(t)*σ(t) is defined. But the code in class SigmaLoss from loss.py,
I see
``` python3
        # sigma = F.relu(raw[...,3] + noise)
        alpha = raw2alpha(raw[...,3] + noise, dists)  # [N_rays, N_samples]
        weights = alpha * torch.cumprod(torch.cat([torch.ones((alpha.shape[0], 1)).to(device), 1.-alpha + 1e-10], -1), -1)[:, :-1]
        
        loss = -torch.log(weights) * torch.exp(-(z_vals - depths[:,None]) ** 2 / (2 * err)) * dists
        loss = torch.sum(loss, dim=1)
        
        return loss
```
`
loss = -torch.log(weights) * torch.exp(-(z_vals - depths[:,None]) ** 2 / (2 * err)) * dists
`
here, the loss function is defined by weights and T_i.
I think that from the paper's h(t) definition, the loss should be defined by 
`
loss = torch.log(sigma) * torch.exp(-(z_vals - depths[:,None]) ** 2 / (2 * err)) * dists
`
but, the actual code is not.
Can I guess why this code is different from the DSNeRF paper?
Additionally, can I ask why the '-' term is added in front of torch.log(weights)?

**Comment by dunbar12138** [comment]
Hi, thanks for your interest!

In the continuous case, 
$$h(t) = T(t)\sigma(t)$$
as 
$$Color = \int h(t) c(t) dt$$.

While in the discrete case,
$$h_t = T_t \alpha_t$$
as
$$Color = \sum h_t c_t$$.

More details could be found in the original NeRF [paper](https://arxiv.org/pdf/2003.08934.pdf).

**Comment by jhq1234** [comment]
Thanks! I can understand why h_t = T_rα_t.
Additionally, can I ask one more thing?

From the paper, depth loss is defined by 
$\sum_k \log h_k \exp \left(-\frac{\left(t_k-\mathbf{D}_{i j}\right)^2}{2 \hat{\sigma}_i^2}\right) \Delta t_k$.

but in code, '-' term is added in front of torch.log(weights) like
$\sum_k \-log h_k \exp \left(-\frac{\left(t_k-\mathbf{D}_{i j}\right)^2}{2 \hat{\sigma}_i^2}\right) \Delta t_k$.

I can't understand this code.
why did this happen?

**Comment by dunbar12138** [comment]
<img width="1477" alt="Screen Shot 2022-12-19 at 16 51 16" src="https://user-images.githubusercontent.com/28395429/208531126-ecfbdb75-f68a-4518-b7fd-f68e2585d504.png">
 Hope this slide helps!

**Comment by Militon** [comment]
> @dunbar12138 Thanks! I can understand why ht=Trαt. Additionally, can I ask one more thing?
> 
> From the paper, depth loss is defined by ∑klog⁡hkexp⁡(−(tk−Dij)22σ^i2)Δtk.
> 
> but in code, '-' term is added in front of torch.log(weights) like ∑k−loghkexp⁡(−(tk−Dij)22σ^i2)Δtk.
> 
> I can't understand this code. why did this happen?

@dunbar12138 Can you answer on this question too?

**Comment by dunbar12138** [comment]
Hi, sorry for the confusion. It is actually a typo in the paper. There should be a '-' in front of the log-likelihood since we want to maximize it.
